### ISSUE TITLE
[Feature request] Ability to log event properties using a Map similar to Mixpanel

### ISSUE THREAD

**Issue by agramian** [amplitude/Amplitude-Android#148]
Mixpanel allows setting event properties by passing a `Map<String, Object> properties` using `set*Map()` methods.  It would be nice to have this same functionality with Amplitude.

**Comment by thsaravana** [comment]
Yes. Construction of a JSONObject is really messy in the client code. It would be better to do the Map -> JSONObject conversion within the `AmplitudeClient`. 

Can't we have similar methods like `logEvent()` that takes `Map` instead of `JSONObject` ?
